mm2src/coins/eth/eth_balance_events.rs

```diff
@@ -113,15 +143,18 @@ async fn fetch_balance(
 }
 
 #[async_trait]
-impl EventBehaviour for EthCoin {
-    fn event_name() -> EventName { EventName::CoinBalance }
-
-    fn error_event_name() -> ErrorEventName { ErrorEventName::CoinBalanceError }
-
-    async fn handle(self, interval: f64, tx: oneshot::Sender<EventInitStatus>) {
-        const RECEIVER_DROPPED_MSG: &str = "Receiver is dropped, which should never happen.";
-
-        async fn start_polling(coin: EthCoin, ctx: MmArc, interval: f64) {
+impl EventStreamer for EthBalanceEventStreamer {
+    type DataInType = NoDataIn;
+
+    fn streamer_id(&self) -> String { format!("BALANCE:{}", self.coin.ticker) }
+
+    async fn handle(
+        self,
+        broadcaster: Broadcaster,
+        ready_tx: oneshot::Sender<Result<(), String>>,
+        _: impl StreamHandlerInput<NoDataIn>,
+    ) {
+        async fn start_polling(streamer_id: String, broadcaster: Broadcaster, coin: EthCoin, interval: f64) {
```

Comment:
I can see that electrum subscriptions are used for utxo . BTW for eth there is also an eth_subscribe rpc to get logs for list of addresses. I guess it may be a todo.

Reply:
I remember having a todo about this somewhere, found it eventually on the SoW doc.
Isn't this a block though: https://github.com/ethereum/go-ethereum/issues/27558